src/SEBT.Portal.Infrastructure/Repositories/InMemoryOtpRepository.cs

```diff
@@ -0,0 +1,44 @@
+using Microsoft.Extensions.Caching.Memory;
+using Sebt.Portal.Core.Models.Auth;
+using SEBT.Portal.Core.Repositories;
+
+namespace SEBT.Portal.Infrastructure.Repositories
+{
+    /// <summary>
+    /// An in-memory implementation of <see cref="IOtpRepository"/> that uses <see cref="IMemoryCache"/> for storing OTP codes.
+    /// </summary>
+    /// <param name="memoryCache">The memory cache instance used to store and retrieve OTP codes.</param>
+    /// <remarks>
+    /// This implementation is suitable for single-instance applications. For distributed scenarios,
+    /// consider using a distributed cache implementation instead.
+    /// </remarks>
+    public class InMemoryOtpRepository(IMemoryCache memoryCache) : IOtpRepository
+    {
+        public async Task SaveOtpCodeAsync(OtpCode newOtpCode)
+        {
+            var existingCode = await GetOtpCodeByEmailAsync(newOtpCode.Email);
+
+            if (existingCode != null)
+            {
+                await DeleteOtpCodeByEmailAsync(newOtpCode.Email);
+            }
+
+            memoryCache.Set(newOtpCode.Email, newOtpCode, newOtpCode.ExpiresAt);
+            return;
+        }
+
+        public Task<OtpCode?> GetOtpCodeByEmailAsync(string email)
```

Comment:
I think this is ultimately a synchronous operation?  Let me know if I'm reading this wrong though